intbot/core/integrations/pretalx.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any
+
+import httpx
+from core.models import PretalxData
+from django.conf import settings
+
+PRETALX_EVENT = "ep2025"
+base_url = f"https://pretalx.com/api/events/{PRETALX_EVENT}/"
+
+RESOURCES = {
+    # Questions need to be passed to include answers in the same endpoint,
+    # saving us later time with joining the answers.
+    PretalxData.PretalxEndpoints.submissions: "submissions?questions=all",
+    PretalxData.PretalxEndpoints.speakers: "speakers?questions=all",
+}
+
+
+JsonType = dict[str, Any]
+
+
+def fetch_pretalx_data(resource) -> list[JsonType]:
+    headers = {
+        "Authorization": f"Token {settings.PRETALX_API_TOKEN}",
+        "Content-Type": "application/json",
+    }
+
+    endpoint = RESOURCES[resource]
+    url = base_url + f"{endpoint}"
+
+    # Pretalx paginates the output, so we will need to do multiple requests and
+    # then merge mutliple pages to one big dictionary
+    res0 = []
+    data = {"next": url}
+    n = 0
+    while url := data["next"]:
+        n += 1
+        response = httpx.get(url, headers=headers)
+
+        if response.status_code != 200:
+            raise Exception(f"Error {response.status_code}: {response.text}")
+
+        data = response.json()
+        res0 += data["results"]
```

Comment:
I find this a bit cryptic:
* can we use `results` instead of `res0`? it's not clear to me what does `0` represent here
* can we use `page` instead of `n`?
* can we use `extend` instead of `+`?
* if we want to fetch the next page, shouldn't it be at the end?

```python
        data = response.json()
        results.extend(data.get("results", []))
        url = data.get("next")
```

Reply:
I changed it a bit and clarified the names of the variables, but I kept the original structure of how the pagination is consumed, and added some comment how it's meant to work.